test/project_types/extension/features/argo_config_test.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'project_types/extension/extension_test_helpers'
+
+module Extension
+  module Features
+    class ArgoConfigTest < MiniTest::Test
+      def setup
+        super
+        File.stubs(:size?).returns(true)
+        ShopifyCli::ProjectType.load_type(:extension)
+      end
+
+      def test_parses_and_symbolizes_yaml_hash
+        value = {}
+        another_value = 10
+        yml = { "value": value, "another_value": another_value }
+        YAML.stubs(:load_file).returns(yml)
+
+        parsed_config = ArgoConfig.parse_yaml(@context, [:value, :another_value])
+
+        assert_equal(value, parsed_config[:value])
+        assert_equal(another_value, parsed_config[:another_value])
+      end
+
+      def test_aborts_when_yaml_is_invalid
+        Psych::SyntaxError.any_instance.stubs(:initialize)
+        YAML.stubs(:load_file).raises(Psych::SyntaxError)
+
+        assert_raises(ShopifyCli::Abort) { ArgoConfig.parse_yaml(@context) }
+      end
+
+      def test_aborts_when_yaml_is_not_a_hash
+        YAML.stubs(:load_file).returns(false)
+
+        assert_raises(ShopifyCli::Abort) { ArgoConfig.parse_yaml(@context) }
+      end
+
+      def test_returns_empty_hash_when_file_not_found_or_empty
```

Comment:
> file not found or empty

In principle I'd have separated those two tests, but I guess `File.size?` covers both cases? If so, carry on 😃 👍

Reply:
It does and since we are stubbing it I didn't see the need to repeat the test